src/System.Drawing.Primitives/src/System/Drawing/Color.cs

```diff
@@ -645,9 +645,8 @@ public override bool Equals(object obj)
 
         public override int GetHashCode()
         {
-            return unchecked(value.GetHashCode() ^
-                    state.GetHashCode() ^
-                    knownColor.GetHashCode());
+            int hash = value.GetHashCode();
+            return unchecked((hash << 5) - hash + (knownColor.GetHashCode() << 3) + state);
```

Comment:
Don't bother with the test case unless you find it interesting in itself (but I thought maybe you might have, and hence have an answer). I'm happy to just go with your approach.